### PR TITLE
Queues.manage_lambdas handles contribute_retry (#2913)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented on demo expectations                        <sub>or labelled PR as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that Demo expectations are clear              <sub>or PR is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and build passes in `sandbox` <sub>or added `no sandbox` label</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [x] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [x] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [x] Purchased BQ slot committment in `dev`                <sub>or this PR does not require reindexing `dev`</sub>
- [x] Started reindex in `dev`                              <sub>or this PR does not require reindexing `dev`</sub>
- [x] Deleted BQ slot committment in `dev`                  <sub>or this PR does not require reindexing `dev`</sub>
- [x] Checked for failures in `dev`                         <sub>or this PR does not require reindexing `dev`</sub>
- [x] Purchased BQ slot committment in `prod`               <sub>or this PR does not require reindexing `prod`</sub>
- [x] Started reindex in `prod`                             <sub>or this PR does not require reindexing `prod`</sub>
- [x] Deleted BQ slot committment in `prod`                 <sub>or this PR does not require reindexing `prod`</sub>
- [x] Checked for failures in `prod`                        <sub>or this PR does not require reindexing `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

File: src/azul/queues.py [hunk 1]
```diff
@@ -434,13 +434,17 @@ def submit(f, *args, **kwargs):
...
+                    submit(self._manage_sqs_push, config.indexer_name + '-contribute_retry', queue, enable)
                 elif queue_name == config.tallies_queue_name():
                     submit(self._manage_sqs_push, config.indexer_name + '-aggregate', queue, enable)
                 elif queue_name == config.tallies_queue_name(retry=True):
                     # FIXME: Brittle coupling between the string literal below and
                     #        the handler function name in app.py
                     #        https://github.com/DataBiosphere/azul/issues/1848
                     submit(self._manage_sqs_push, config.indexer_name + '-aggregate_retry', queue, enable)
+                else:
+                    assert False
```
Comment (danielsotirhos) [inline src/azul/queues.py:447]:
```suggestion
                    assert False, queue_name
```
However, the assertion can't be used as the `if` block doesn't cover all queue names:
```
(azul2) daniel@Crispin ~/azul2 $ python scripts/manage_queues.py purge_all
2021-03-23 10:18:30,818 INFO    MainThread: Found credentials in environment variables.
2021-03-23 10:18:31,785 INFO    ThreadPoolExecutor-0_2: Found credentials in environment variables.
2021-03-23 10:18:31,788 INFO    ThreadPoolExecutor-0_0: Found credentials in environment variables.
2021-03-23 10:18:31,789 INFO    ThreadPoolExecutor-0_3: Found credentials in environment variables.
2021-03-23 10:18:31,790 INFO    ThreadPoolExecutor-0_1: Found credentials in environment variables.
2021-03-23 10:18:31,791 INFO    ThreadPoolExecutor-0_4: Found credentials in environment variables.
2021-03-23 10:18:32,399 WARNING ThreadPoolExecutor-0_0: azul-indexer-daniel is already disabled.
2021-03-23 10:18:32,428 INFO    ThreadPoolExecutor-0_3: Push from 'https://queue.amazonaws.com/122796619775/azul-tallies-daniel.fifo' to lambda function 'azul-indexer-daniel-aggregate' is in state 'Disabled'.
2021-03-23 10:18:32,705 INFO    ThreadPoolExecutor-0_2: Push from 'https://queue.amazonaws.com/122796619775/azul-notifications_retry-daniel' to lambda function 'azul-indexer-daniel-contribute_retry' is in state 'Disabled'.
2021-03-23 10:18:32,712 INFO    ThreadPoolExecutor-0_1: Push from 'https://queue.amazonaws.com/122796619775/azul-notifications-daniel' to lambda function 'azul-indexer-daniel-contribute' is in state 'Disabled'.
2021-03-23 10:18:32,743 INFO    ThreadPoolExecutor-0_4: Push from 'https://queue.amazonaws.com/122796619775/azul-tallies_retry-daniel.fifo' to lambda function 'azul-indexer-daniel-aggregate_retry' is in state 'Disabled'.
Traceback (most recent call last):
  File "scripts/manage_queues.py", line 93, in <module>
    main(sys.argv[1:])
  File "scripts/manage_queues.py", line 74, in main
    queues.purge_all()
  File "/Users/daniel/azul2/src/azul/queues.py", line 353, in purge_all
    self.purge_queues_safely(self.azul_queues())
  File "/Users/daniel/azul2/src/azul/queues.py", line 356, in purge_queues_safely
    self.manage_lambdas(queues, enable=False)
  File "/Users/daniel/azul2/src/azul/queues.py", line 447, in manage_lambdas
    assert False, queue_name
AssertionError: azul-tallies_fail-daniel.fifo
```

Comment (danielsotirhos) [inline src/azul/queues.py:447]:
```suggestion
                    assert False, queue_name
```
However the assertion can't be used as the `if` block doesn't cover all possible queues:
```
(azul2) daniel@Crispin ~/azul2 $ python scripts/manage_queues.py purge_all
2021-03-23 10:18:30,818 INFO    MainThread: Found credentials in environment variables.
2021-03-23 10:18:31,785 INFO    ThreadPoolExecutor-0_2: Found credentials in environment variables.
2021-03-23 10:18:31,788 INFO    ThreadPoolExecutor-0_0: Found credentials in environment variables.
2021-03-23 10:18:31,789 INFO    ThreadPoolExecutor-0_3: Found credentials in environment variables.
2021-03-23 10:18:31,790 INFO    ThreadPoolExecutor-0_1: Found credentials in environment variables.
2021-03-23 10:18:31,791 INFO    ThreadPoolExecutor-0_4: Found credentials in environment variables.
2021-03-23 10:18:32,399 WARNING ThreadPoolExecutor-0_0: azul-indexer-daniel is already disabled.
2021-03-23 10:18:32,428 INFO    ThreadPoolExecutor-0_3: Push from 'https://queue.amazonaws.com/122796619775/azul-tallies-daniel.fifo' to lambda function 'azul-indexer-daniel-aggregate' is in state 'Disabled'.
2021-03-23 10:18:32,705 INFO    ThreadPoolExecutor-0_2: Push from 'https://queue.amazonaws.com/122796619775/azul-notifications_retry-daniel' to lambda function 'azul-indexer-daniel-contribute_retry' is in state 'Disabled'.
2021-03-23 10:18:32,712 INFO    ThreadPoolExecutor-0_1: Push from 'https://queue.amazonaws.com/122796619775/azul-notifications-daniel' to lambda function 'azul-indexer-daniel-contribute' is in state 'Disabled'.
2021-03-23 10:18:32,743 INFO    ThreadPoolExecutor-0_4: Push from 'https://queue.amazonaws.com/122796619775/azul-tallies_retry-daniel.fifo' to lambda function 'azul-indexer-daniel-aggregate_retry' is in state 'Disabled'.
Traceback (most recent call last):
  File "scripts/manage_queues.py", line 93, in <module>
    main(sys.argv[1:])
  File "scripts/manage_queues.py", line 74, in main
    queues.purge_all()
  File "/Users/daniel/azul2/src/azul/queues.py", line 353, in purge_all
    self.purge_queues_safely(self.azul_queues())
  File "/Users/daniel/azul2/src/azul/queues.py", line 356, in purge_queues_safely
    self.manage_lambdas(queues, enable=False)
  File "/Users/daniel/azul2/src/azul/queues.py", line 447, in manage_lambdas
    assert False, queue_name
AssertionError: azul-tallies_fail-daniel.fifo
```

**Review by danielsotirhos — CHANGES_REQUESTED** [review]

**Review by danielsotirhos — APPROVED** [review]
LGTM

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
For demo, run `reindex.py` and, using log output, show that contribution retry queue is being reset and Lambda push disabled/reenabled for that queue.

**Pull request merged**